src/contracts/gen/BatchExchange.ts

```diff
@@ -105,41 +68,97 @@ export interface BatchExchange extends Contract {
       3: string
     }>
 
-    cancelOrders(ids: (number | string)[]): TransactionObject<void>
-
-    acceptingSolutions(batchIndex: number | string): TransactionObject<boolean>
-
-    getEncodedAuctionElements(): TransactionObject<string>
+    getPendingDeposit(
+      user: string,
+      token: string,
+    ): TransactionObject<{
+      0: string
+      1: string
+    }>
 
-    addToken(token: string): TransactionObject<void>
+    getPendingWithdraw(
+      user: string,
+      token: string,
+    ): TransactionObject<{
+      0: string
+      1: string
+    }>
 
     getBalance(user: string, token: string): TransactionObject<string>
 
-    freeStorageOfOrder(ids: (number | string)[]): TransactionObject<void>
-
     BATCH_TIME(): TransactionObject<string>
 
     getCurrentBatchId(): TransactionObject<string>
 
-    tokenAddressToIdMap(addr: string): TransactionObject<string>
-
     requestFutureWithdraw(token: string, amount: number | string, batchId: number | string): TransactionObject<void>
 
     hasValidWithdrawRequest(user: string, token: string): TransactionObject<boolean>
 
     MAX_TOKENS(): TransactionObject<string>
 
-    getPendingDepositBatchNumber(user: string, token: string): TransactionObject<string>
-
     withdraw(user: string, token: string): TransactionObject<void>
 
     MAX_TOUCHED_ORDERS(): TransactionObject<string>
 
+    addToken(token: string): TransactionObject<void>
+
+    placeOrder(
+      buyToken: number | string,
+      sellToken: number | string,
+      validUntil: number | string,
+      buyAmount: number | string,
+      sellAmount: number | string,
+    ): TransactionObject<string>
+
+    placeValidFromOrders(
+      buyTokens: (number | string)[],
+      sellTokens: (number | string)[],
+      validFroms: (number | string)[],
+      validUntils: (number | string)[],
+      buyAmounts: (number | string)[],
+      sellAmounts: (number | string)[],
+    ): TransactionObject<string[]>
+
+    cancelOrders(ids: (number | string)[]): TransactionObject<void>
+
+    replaceOrders(
+      cancellations: (number | string)[],
+      buyTokens: (number | string)[],
+      sellTokens: (number | string)[],
+      validFroms: (number | string)[],
+      validUntils: (number | string)[],
+      buyAmounts: (number | string)[],
+      sellAmounts: (number | string)[],
+    ): TransactionObject<string[]>
+
+    submitSolution(
+      batchIndex: number | string,
+      claimedObjectiveValue: number | string,
+      owners: string[],
+      orderIds: (number | string)[],
+      buyVolumes: (number | string)[],
+      prices: (number | string)[],
+      tokenIdsForPrice: (number | string)[],
+    ): TransactionObject<string>
+
+    tokenAddressToIdMap(addr: string): TransactionObject<string>
+
+    tokenIdToAddressMap(id: number | string): TransactionObject<string>
+
+    hasToken(addr: string): TransactionObject<boolean>
+
+    getEncodedUserOrders(user: string): TransactionObject<string>
```

Comment:
Probably the reason why https://github.com/gnosis/dex-react/pull/327 is failing. That ABI doesn't have this method yet.

Reply:
yep :) 

Anyways, it can still fail. There's a scalability issue. I've tagged you in a relevant issue in dex-contracts to follow the matter